### PR TITLE
Update language throughout the plugin from Group to Team

### LINKED ISSUE #441: Update plugin description and plugin author from "Group" to "Team"
Now that we're an official core WP Team, we should update the language in the plugin description and the plugin author name to reflect this. Currently:

image

### PR DESCRIPTION
**Pull request by mxbclang**
## Summary

Fixes #441 

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

**Review by felixarntz — APPROVED** [review]
Thank @bethanylang!

We may also want to replace the entry point links for the Performance Team to be `https://make.wordpress.org/performance/` instead of `https://make.wordpress.org/core/tag/performance/` (the latter is still relevant, but no longer as the main entry point). We can probably wait a bit more with that though, until that site has some content.

**Review by jeffpaul — APPROVED** [review]
Looks to be all the spots I see to update as well, nice update!

File: docs/Proposing-a-new-module.md [hunk 1]
```diff
@@ -18,10 +18,10 @@ Proposed or upcoming WordPress core features are also called “[feature project
...
 
 ## Requirements for a new module
-Based on the above lifecycle for a module, there are a few initial requirements for proposing a new module. The performance team would like to ensure that modules inside the Performance Lab plugin keep moving forward over time, and the requirements outlined here help set the foundation.
+Based on the above lifecycle for a module, there are a few initial requirements for proposing a new module. The Performance Team would like to ensure that modules inside the Performance Lab plugin keep moving forward over time, and the requirements outlined here help set the foundation.
```
Comment (mukeshpanchal27) [inline docs/Proposing-a-new-module.md:24]: Can we use lowercase text for `Performance Team` as we used before?

**Review by mukeshpanchal27 — APPROVED** [review]
LGTM. One question regarding the text case.

File: docs/Proposing-a-new-module.md (hunk 1 above)
Comment (mxbclang) [reply docs/Proposing-a-new-module.md:24]: @mukeshpanchal27 I think having it lowercase before was actually an oversight; we should always use the capital case Performance Team (or previously, Performance Group).

Comment (mukeshpanchal27) [reply docs/Proposing-a-new-module.md:24]: OK, Thanks for the clerification.

**Comment by mxbclang** [comment]
@felixarntz Circling back on this one as I noticed these language updates again with today's release.  :)

**Comment by felixarntz** [comment]
@bethanylang There was still my feedback from https://github.com/WordPress/performance/pull/442#pullrequestreview-1042105714 which was missing, but I just added this now in https://github.com/WordPress/performance/pull/442/commits/7fedc96fdecd0865309ff536188c187817b0ec35. This way we point users to the correct place `https://make.wordpress.org/performance/` instead of `https://make.wordpress.org/core/tag/performance/` when referring to the Performance Team.

**Review by felixarntz — APPROVED** [review]

**Pull request merged**